packages/hoppscotch-cli/src/utils/request.ts

```diff
@@ -0,0 +1,336 @@
+
+        // !NOTE: Temporary `config.supported` check
+        if ((err.config as RequestConfig).supported === false) {
+          status = 501;
+          statusText = responseErrors[status];
+        } else if (!err.response) {
+          status = 408;
+          statusText = responseErrors[status];
+        } else {
+          status = err.response.status;
+          statusText = err.response.statusText;
+        }
+      } else {
+        status = 600;
+        statusText = responseErrors[status];
+      }
+      runnerResponse.status = status;
+      runnerResponse.statusText = statusText;
+
+      return runnerResponse;
+    }
+  };
+
+/**
+ * The request parser from the collection JSON.
+ * @param collection The collection object parsed from the JSON.
+ * @param rootPath The folder path.
+ * @returns RequestStack[] - Created request-stacks successfully parsed
+ * from HoppCollection.
+ * HoppCLIError - On error while parsing HoppCollection.
+ */
+export const requestsParser = (
+  collection: HoppCollection<HoppRESTRequest>,
+  rootPath: string = "$ROOT"
+): TE.TaskEither<HoppCLIError, RequestStack[]> =>
+  pipe(
+    /**
+     * Mapping collection's HoppRESTRequests to EffectiveHoppRESTRequest,
+     * then running preRequestScriptRunner over them and mapping output
+     * to generate RequestStack[] which is binded to PARSED_REQUESTS.
+     */
+    collection.requests,
+    A.map(
+      (hoppRequest) =>
+        Object({
+          ...hoppRequest,
+          effectiveFinalBody: null,
+          effectiveFinalHeaders: [],
+          effectiveFinalParams: [],
+          effectiveFinalURL: S.empty,
+        }) as EffectiveHoppRESTRequest
+    ),
+    A.map(preRequestScriptRunner),
+    TE.sequenceArray,
+    TE.map(RA.toArray),
+    TE.map(
+      A.map((effHoppRequest) =>
+        createRequest(`${rootPath}/${collection.name}`, effHoppRequest)
+      )
+    ),
+    TE.bindTo("PARSED_REQUESTS"),
+
+    /**
+     * Recursive call to requestsParser on collections's folder, the
+     * RequestStack[] output concated with PARSED_REQUESTS, and the
+     * overall output is binded to FINAL_PARSED_REQUESTS.
+     */
+    TE.chain(({ PARSED_REQUESTS }) =>
+      pipe(
+        collection.folders,
+        A.map((a) => requestsParser(a, `${rootPath}/${collection.name}`)),
+        TE.sequenceArray,
+        TE.map(RA.toArray),
+        TE.map(A.flatten),
+        TE.map(A.concat(PARSED_REQUESTS))
+      )
+    ),
+    TE.bindTo("FINAL_PARSED_REQUESTS"),
+    TE.map(({ FINAL_PARSED_REQUESTS }) => FINAL_PARSED_REQUESTS)
+  );
+
+const preRequestScriptRunner = (
+  request: EffectiveHoppRESTRequest
+): TE.TaskEither<HoppCLIError, EffectiveHoppRESTRequest> =>
+  pipe(
+    TE.of(request),
+    TE.map((req) => req.preRequestScript),
+    TE.chain((preScript) =>
+      runPreRequestScript(preScript, { global: [], selected: [] })
+    ),
+    TE.map((envs) => Object({ name: "Env", variables: envs.selected })),
```

Comment:
Using the object constructor makes this object lose type information (this TE now returns an `any`)

```suggestion
      TE.map((envs) => ({ name: "Env", variables: envs.selected }))
```